### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by jkolyakov**
# Description
Integration to main. Added all frontend features except the navbar buttons. Backend added transcript, QAs REST APIs and a dummy transcript.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Integration

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manual Testing

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
Most tests do not apply here

### CODE REVIEW

File: mern/client/src/screens/Landing/landing.js [hunk 1]
```diff
@@ -1,55 +1,39 @@
...
-      <div className="container">
-        <h1 className="h1 title">
-          Transcript to chatbot <br /> with a couple clicks
-        </h1>
-        <h4 className="subtitle">A flow-building plugin for Voiceflow</h4>
-        <div className="buttonContainer">
-          <button className="button">Upload Transcript</button>
-          <button className="button">Recover Session</button>
-        </div>
+import GenericButton from "../../Components/Buttons/GenericButton";
```
Comment (TerryTian21) [inline mern/client/src/screens/Landing/landing.js:3]: Tested this and seems to work good.

File: mern/client/src/screens/StartingIntent/StartingIntent.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import React from "react";
```
Comment (TerryTian21) [inline mern/client/src/screens/StartingIntent/StartingIntent.js:1]: This is a good start to the start intent screen.

There are currently place-holders for the question and intent. Remember to have the back-end call to get the manipulated/parsed transcript data to be displayed instead.

File: mern/client/src/screens/UploadTranscript/SuccessModal.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                    onClick={() => closeModal(false)}
+                >
+                    Cancel
+                </button>
+                <button
+                    className="text-green-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
+                    type="button"
+                    onClick={() => {
+                        closeModal(false);
+                        PageChange("/");
```
Comment (TerryTian21) [inline mern/client/src/screens/UploadTranscript/SuccessModal.jsx:39]: Here the page change function should go to the StartingIntent page.

Have a Axios call to create flow, which will then return the initial values to be displayed.

File: mern/client/src/withRouter.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { useNavigate } from 'react-router-dom';
```
Comment (TerryTian21) [inline mern/client/src/withRouter.js:1]: This should be organized into a different folder to make the code cleaner. Maybe not just in SRC, put it in the generic buttons folder or even the utils folder.

**Review by TerryTian21 — APPROVED** [review]
src/fonts folder is structured weirdly. There exists a Montserrat folder that contains the Nunito Fonts. Could not make a comment directly on the file since it is not supported.

Overall, the changes didn't seem to break anything, just need to make sure to add functionality to frontend components and complete the backend required for parsing the input transcript.

File: mern/client/src/screens/RecoverSession/recoverSession.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            onClick={() => this.handleChange}
+            disabled={false}
+            text={"Begin Session"}
+          />
+        </form>
+        <div className="buttonContainer">
+          <GenericButton
+            buttonType="outline"
+            onClick={this.handleBack}
+            disabled={false}
```
[the remark this answers is not in the document]
Comment (jkolyakov) [reply mern/client/src/screens/RecoverSession/recoverSession.js:54]: Will be done in a different PR

File: mern/client/src/screens/StartingIntent/StartingIntent.js (hunk 2 above)
Comment (jkolyakov) [reply mern/client/src/screens/StartingIntent/StartingIntent.js:1]: Noted and added to Trello board as an issue

File: mern/client/src/screens/UploadTranscript/uploadTranscript.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+          />
+          Choose File
+        </button>
+      </div>
+      <h4 className="subtitle"> {fileName} </h4>
+      <div className="buttonContainer2">
+        <GenericButton
+          buttonType={files ? "blue" : "disabled"}
+          onClick={() => {
+            transcriptJSONConverter(fileName, files); // Checks if the transcript is a string, and then sends transcript to DB
```
[the remark this answers is not in the document]
Comment (jkolyakov) [reply mern/client/src/screens/UploadTranscript/uploadTranscript.js:69]: Known feature that we need to add. Already a card on the trello board

File: mern/client/src/utils/transcript.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+var res
+
+// Sends Transcript to Database
+async function uploadFile(transcript) {
+
+  axios
+    .post("http://localhost:5000/transcript/add", transcript)
+    .then(() => {
+      console.log("Transcript Uploaded!");
+      return true
```
[the remark this answers is not in the document]
Comment (jkolyakov) [reply mern/client/src/utils/transcript.js:12]: See response above

**Pull request merged**